sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/CommunicationTokenCredentialTests.java

```diff
@@ -126,45 +137,92 @@ public void refresherShouldBeCalledImmediatelyWithExpiredToken()
         immediateFresher.resetCallCount();
         CountDownLatch countDownLatch = new CountDownLatch(1);
         immediateFresher.setOnCallReturn(countDownLatch::countDown);
-        CommunicationTokenRefreshOptions tokenRefreshOptions = new CommunicationTokenRefreshOptions(immediateFresher, true, tokenStr);
+        CommunicationTokenRefreshOptions tokenRefreshOptions = new CommunicationTokenRefreshOptions(immediateFresher)
+                .setRefreshProactively(true)
+                .setInitialToken(tokenStr);
+        CommunicationTokenCredential tokenCredential = new CommunicationTokenCredential(tokenRefreshOptions);
+
+        countDownLatch.await();
+        assertEquals(1, immediateFresher.numCalls());
+        StepVerifier.create(tokenCredential.getToken())
+                .assertNext(token ->
+                        assertFalse(token.isExpired(), "Refreshable AccessToken should not expire after refresh"))
+                .verifyComplete();
+        tokenCredential.close();
+    }
+
+    @Test
+    public void fractionalBackoffAppliedWhenTokenExpiring()
+            throws InterruptedException, ExecutionException, IOException {
+        int validForSecs = 8;
+        double expectedTotalCallsTillLastSecond = Math.floor(Math.log(validForSecs));
+        String tokenStr = tokenMocker.generateRawToken("resourceId", "userIdentity", 8);
+        AccessToken accessToken = (new TokenParser()).parseJWTToken(tokenStr);
+        immediateFresher.resetCallCount();
+        immediateFresher.setRefreshedToken(tokenStr);
+        CommunicationTokenRefreshOptions tokenRefreshOptions = new CommunicationTokenRefreshOptions(immediateFresher)
+                .setRefreshProactively(true)
+                .setInitialToken(tokenStr);
+        CommunicationTokenCredential tokenCredential = new CommunicationTokenCredential(tokenRefreshOptions);
+        long tokenTtlSecs = accessToken.getExpiresAt().toInstant().getEpochSecond() - Instant.now().getEpochSecond();
+        while (tokenTtlSecs > 1) {
+            tokenTtlSecs = accessToken.getExpiresAt().toInstant().getEpochSecond() - Instant.now().getEpochSecond();
+        }
```

Comment:
```suggestion
        long tokenTtlSecs = 0;
        do {
            tokenTtlSecs = expirationTime - Instant.now().getEpochSecond();
        } while (tokenTtlSecs > 1);
```

Reply:
- the test takes quite long to complete. can we:
  -  use a simplified version of it, similarly to e.g. [python](https://github.com/Azure/azure-sdk-for-python/pull/21834/files#diff-171e016d8fe5cb1e0c5886700e11877493940a3f14f62de59857b31adda82590)
  - or somehow utilize the `StepVerifier` and `CountDownLatch`?
- is that the reason why `validForSecs` is set to 8?